fhireval/data/__init__.py

```diff
@@ -51,30 +173,91 @@ def delete_content_for_resource(host, resource_type, records=None):
+                if return_code == 409:
+                    redos.append(response)
+                else:
+                    deleted_count += 1
+        
+        for response in redos:
+            delete_from_response(host, resource_type, response, sleep_til_gone=False)
+            deleted_count += 1
+        
+        if deleted_count > 0 and sleep_til_gone:
+            host.sleep_until(f"{resource_type}", 0, message=resource_type)
+        elif len(entries) > 0:
+            print()
     else:
         # We may need slightly different ways to delete some of the resource types
         for record in records:
-            if 'identifier' in record:
-                for response in host.get(
-                    f"{resource_type}?identifier={record['identifier'][0]['system']}|{record['identifier'][0]['value']}"
-                ).entries:
-                    if "resource" in response:
-                        delete_from_response(response)
+            identifier = get_identifier(record)
+
+            qry = f"identifier={identifier[0]['system']}|{identifier[0]['value']}"
+            if identifier is None and resource_type in _alternate_search_params:
+                if param not in record:
+                    print(record.keys())
+                qry = f"{param}={record[param]}"
+            for response in host.get(
+                f"{resource_type}?{qry}"
+            ).entries:
+                if "resource" in response:
+                    delete_from_response(host, resource_type, response)
+
+def resource_order():
+    return _resource_order
 
-def load_test_data():
-    global _resource_data
+def load_test_model(host):
+    """Load the full model into memory"""
+
+    print("Loading Model into memory")
+    wait_list = []
+    for resource in _model_resource_order:
+        if resource in _resource_data['_MODEL']:
+            for record in _resource_data['_MODEL'][resource]:
+                response = host.load(resource, record, validate_only=False, skip_insert_if_present=True)
+
+                if response['status_code'] != 201:
+                    pdb.set_trace()
+                    print(resource)
+                    print(response)
+                wait_list.append(f"{resource}?url={record['url']}")
+    print("Sleeping until model components are recognized")
+    for url in wait_list:
+        print(url)
+        host.sleep_until(url, 1, message=url)
+
+
+def delete_test_model(host):
+    """Purges the existing customized model in order to permit a newer version to be loaded in it's place"""
+    print("Deleting the previous test model if present")
+
+    wait_list = []
+    for resource in _model_resource_order[::-1]:
+        if resource in _resource_data['_MODEL']:
+            for record in _resource_data['_MODEL'][resource]:
+                print(f"Deleting {resource} by name: {record['name']}")
+                delete_content_by_name(host, resource, record, sleep_til_gone=False)

```

Comment:
We don't check response status here?

Reply:
That's a good catch. Unlike some others, this return is an array, so I need to give how to handle that some thought.